polygon/bridge/db.go

```diff
@@ -165,11 +155,9 @@ func (s *MdbxStore) GetSprintLastEventID(ctx context.Context, lastID uint64, tim
 		}
 
 		eventID = event.ID
-
-		if bytes.Equal(k, kDBLast) {
-			return eventID, nil
-		}
 	}
+
+	return eventID, nil
```

Comment:
Shouldn't you return something to indicate the event was not found here? This the case where all the events are after the timeLimit right?

Reply:
It's handled in the calling function [here](https://github.com/ledgerwatch/erigon/blob/af719e3035ed096013a60a3166ed838ca75d8086/polygon/bridge/bridge.go#L140)

Reply:
That seems to check something different, no ?   It is checking : `if lastDBID != 0 && lastDBID > b.lastProcessedEventID ` , but `lastBID > b.lastProcessedEventID ` is almost guaranteed by the iteration in `GetSprintLastEventID` which starts from `b.lastProcessedEventID` and going up.

I was talking about the situation where every event you iterate has  `event.Time >= timeLimit`. In that case you should have an EventNotFoundError.

Reply:
That just means we have no events for that block. It should not be an error.

Reply:
In that case I would suggest to return also a bool to indicate if an event was found or not, so as not to break the contract in the docstring of `GetSprintLastEventID` .